app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -173,18 +177,20 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
         val playbackPositionMillis = player.currentPosition
         if (player.playbackState != Player.STATE_ENDED) {
             webappFunctionChannel.exoPlayerUpdateProgress(playbackPositionMillis)
-            apiClient.reportPlaybackProgress(PlaybackProgressInfo().apply {
-                itemId = mediaSource.id
-                canSeek = true
-                isPaused = !player.isPlaying
-                isMuted = false
-                positionTicks = playbackPositionMillis * Constants.TICKS_PER_MILLISECOND
-                val stream = AudioManager.STREAM_MUSIC
-                val volumeRange = audioManager.getVolumeRange(stream)
-                val currentVolume = audioManager.getStreamVolume(stream)
-                volumeLevel = (currentVolume - volumeRange.first) * 100 / volumeRange.width
-                repeatMode = ApiRepeatMode.RepeatNone
-            })
+
+            val stream = AudioManager.STREAM_MUSIC
+            val volumeRange = audioManager.getVolumeRange(stream)
+            val currentVolume = audioManager.getStreamVolume(stream)
+            playStateApi.reportPlaybackProgress(PlaybackProgressInfo(
+                itemId = UUID.fromString(mediaSource.id.replace("^([a-z\\d]{8})([a-z\\d]{4})([a-z\\d]{4})([a-z\\d]{4})([a-z\\d]{12})\$".toRegex(), "$1-$2-$3-$4-$5")), // TODO Use UUIDSerializer
+                canSeek = true,
+                isPaused = !player.isPlaying,
+                isMuted = false,
+                positionTicks = playbackPositionMillis * Constants.TICKS_PER_MILLISECOND,
+                volumeLevel = (currentVolume - volumeRange.first) * 100 / volumeRange.width,
+                repeatMode = RepeatMode.REPEAT_NONE,
+                playMethod = PlayMethod.DIRECT_PLAY, // mediaSource.playMethod is incompatible because it's a string
```

Comment:
So ideally we'd use mediasource.playmethod so we report this correctly but it's a string so we need to add a when statement or something, might be possible to just bug @crobibero about it so it's fixed in the next server release :p

Reply:
actually mediasource is not a server model, my bad

Reply:
For this one we need to change the `JellyfinMediaSource` model to read the `playMethod` json element as the `PlayMethod` enum instead of string, the json parser used doesn't support enums though. We should migrate it to kotlinx

Reply:
Fixed.